### ISSUE TITLE
Passing values to URI

### ISSUE THREAD

**Issue by carlskii** [stajkowski/webhooks#5]
Here's a slightly different use case.

I need to create a POST that looks like this:

```ruby
webhooks_request "Get data from TPP" do
  uri "172.16.28.200/vedsdk/certificates/retrieve?apikey=" + node["tpp"]["apikey"]
  #uri "127.0.0.1/"
  #uri_port 5000
  ssl_validation false
  use_ssl true
  #follow_redirect true
  post_data (
                { 'CertificateDN' => "\\VED\\Policy\\EMV Certificates\\NsCVcQg4fd", 'Format' => "Base64", 'Password' => "lexicon", 'IncludePrivateKey' => "true"}
            )
  header_data (
                  { 'content-type' => 'application/json'}
              )
  expected_response_codes [ 200, 201, 400, 401 ]
  action :post
end
```

So basically I need to pass in the API key as a parameter to the next request. However I don't think the `uri` key supports this:

```
TypeError
---------
no implicit conversion of nil into String
```

**Comment by carlskii** [comment]
Hey there, any idea why this is not working? I think If I crack this I'll be good to go :)

**Comment by stajkowski** [comment]
Yea, np, I'll get this going for you today!

**Comment by carlskii** [comment]
Excellent thanks. Assume its not doing something stupid.

**Comment by carlskii** [comment]
So I made some progress here. I changed the post case action in the provider to look like this so the entire uri gets sent. 

```ruby
when "post"
          Chef::Log.info "Setting up action #{ action }."
          # Changed so that the full uri including query parameters are sent
          #req = Net::HTTP::Post.new(uri.path)   #let's post
          req = Net::HTTP::Post.new(uri)   #let's post
```

However the issue now is that the value of my `default["tpp"]["apikey"] ` attribute is not available in here: 

```ruby
webhooks_request "Get data from TPP" do
  uri "127.0.0.1/vedsdk/certificates/retrieve?apikey=#{node["tpp"]["apikey"]}"
  ssl_validation false
  use_ssl false
  post_data (
                { 'DN' => "NsCVcQg4fd", 'Format' => "Base64", 'Password' => "123", 'Key' => "true"}
            )
  header_data (
                  { 'content-type' => 'application/json', 'auth' => node["tpp"]["username"]}
              )
  expected_response_codes [ 200, 201, 400, 401 ]
  action :post
end
```

However the `["tpp"]["username"]` attribute is. 

I verified the the `node.overide["tpp"]["apikey"]` is working as defined here

```ruby
ruby_block "Extract the TPP API Key" do
  block do
    jsonData = JSON.parse(node["webhooks"]["post_response"])
    jsonData.each do | k, v |
      if k == 'APIKey'
        node.overide["tpp"]["apikey"] = v
      end
    end
  end
  action :run
end
```

 using this: 

```ruby
ruby_block "Print API Key" do
  block do
    print "\nThis should be the key: " + node["tpp"]["apikey"] + "\n"
  end
  action :run
end
```
Which indeed works perfectly, from within the `ruby_block`. So its strange why its empty from within the `webhooks_request`.

**Comment by carlskii** [comment]
OK so I've got this to work now. Basically I have to do this `uri lazy {"127.0.0.1/sdk/data/retrieve?apikey=#{node["tpp"]["apikey"]}"}` in order to get the value of the API key from the attribute: 

```ruby
webhooks_request "Get data from TPP" do
  uri lazy {"127.0.0.1/sdk/data/retrieve?apikey=#{node["tpp"]["apikey"]}"}
  post_data (
                { 'data' => "NsCVcQg4fd"
            )
  header_data (
                  { 'content-type' => 'application/json', 'auth' => "#{lazy {node["server"]["apikey"]} }" }
              )
  expected_response_codes [ 200, 201, 400, 401 ]
  action :post
end
```

Seems weird and a bit hacky to me though but it works. I was under the impression that the all of the attributes would be available directly once they are set.

**Comment by stajkowski** [comment]
Should be able to get this in by Monday, major work being done.

**Comment by carlskii** [comment]
Excellent that will be great.

**Comment by stajkowski** [comment]
Fixed in v0.1.3
